sdk/lib/_internal/js_dev_runtime/private/js_string.dart

```diff
@@ -83,6 +83,104 @@ final class JSString extends Interceptor
     return stringReplaceAllFuncUnchecked(this, from, onMatch, onNonMatch);
   }
 
+  @notNull
+  Iterable<T> splitMap<T>(
+    Pattern pattern, {
+    T Function(Match match)? onMatch,
+    T Function(String nonMatch)? onNonMatch,
+  }) {
+    return stringSplitMapUnchecked(this, pattern, onMatch, onNonMatch);
+  }
+
+  Iterable<T> stringSplitMapUnchecked<T>(String receiver, Pattern pattern,
+      T Function(Match)? onMatch, T Function(String)? onNonMatch) {
+    onMatch ??= (match) => match[0]! as T;
+    onNonMatch ??= (string) => string as T;
```

Comment:
Definitely not sound. If the parameters are optional, I'd emit nothing for a match/non-match if `onMatch`/`onNonMatch` is `null`.
(And I'd probably make them named parameters today.)